src/node_crypto.cc

```diff
@@ -1025,6 +1029,55 @@ void SecureContext::LoadPKCS12(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+#ifndef OPENSSL_NO_ENGINE
+void SecureContext::SetClientCertEngine(
+  const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  CHECK(args.Length() == 1 && args[0]->IsString());
+
+  SecureContext* sc = Unwrap<SecureContext>(args.This());
+
+  // If an engine was previously set, free it.
+  if (sc->ctx_->client_cert_engine != NULL) {
+    ENGINE* e = sc->ctx_->client_cert_engine;
+    sc->ctx_->client_cert_engine = NULL;
+    ENGINE_free(e);
+  }
+
+  const node::Utf8Value engine_id(env->isolate(), args[0]);
+  ENGINE* engine = ENGINE_by_id(*engine_id);
+
+  // Engine not found, try loading dynamically
+  if (engine == nullptr) {
```

Comment:
I believe we have a similar code somewhere in this file. Perhaps, it may be a good idea to split it up and share between two places?